Controller/Adminhtml/Importcategory/Save.php

```diff
@@ -71,147 +79,191 @@ public function __construct(
+                                        
+                                        if ($cateItem->getId()) {
+                                            $existingCategories[$cateItem->getId()] = $cateItem->getName();
+                                            $existingCatPath[$cateItem->getId()] = $cateItem->getPath();
+                                            $existingCatPathName[$cateItem->getId()] = $categoryItem[$categoriesKey];
                                         }
                                     }
                                 }
-                            } elseif (isset($categoryidkey) && ($categoryidkey!='' || $categoryidkey===0) && $storedata!='') {
+                            } else if (isset($categoryIdKey) &&
+                                        ($categoryIdKey != '' || $categoryIdKey === 0) && $storeData != '') {
                                 //update categories
-                                $catemodel = $this->_categoryFactory->create();
-                                if (!$singlestoremode && isset($_stores[$categoryitem[$storedata]])) {
-                                        $catemodel->setStoreId($_stores[$categoryitem[$storedata]]);
+                                $cateModel = $this->_categoryFactory->create();
+
+                                if (!$singleStoreMode && isset($_stores[$categoryItem[$storeData]])) {
+                                    $cateModel->setStoreId($_stores[$categoryItem[$storeData]]);
                                 } else {
-                                    $catemodel->setStoreId($storeId);
+                                    $cateModel->setStoreId($storeId);
                                 }
-                                $cateitem = $catemodel->load($categoryitem[$categoryidkey]);
-                                $nocategoryfound = true;
-                                if ($cateitem->getId()) {
-                                    $nocategoryfound = false;
-                                    $attributesetid = $cateitem->getAttributeSetId();
-                                    $_parentid = $cateitem->getParentId();
+                                $cateItem = $cateModel->load($categoryItem[$categoryIdKey]);
+                                $noCategoryFound = true;
+
+                                if ($cateItem->getId()) {
+                                    $noCategoryFound = false;
+                                    $attributeSetId = $cateItem->getAttributeSetId();
+                                    $categoryId = $cateItem->getParentId();
+
                                     foreach ($cat_data as $key => $value) {
-                                        if (!in_array($key, ['url_key','category_id','url_path','path','level','children_count','full_path'])) {
-                                            $cateitem->setData($key, $value);
+
+                                        if (!in_array($key,
```

Comment:
please refactor this

Reply:
better will be to put these options into a parameter named `$acceptedKeys` or something similar

Reply:
![image](https://user-images.githubusercontent.com/22006850/45302994-a9ee4580-b50c-11e8-8717-e1fc916bcd7b.png)